### DISCUSSION TITLE
mojo packages announcement (A Mojo package is defined by placing an __init__.mojo or __init__.🔥 within a directory)

### DISCUSSION THREAD

**Discussion by guidorice** [modular/modular#486] [Q&A] [answered]
Hi, it looks like mojo packages (basic support for importing) was announced on [2023-07-20](https://docs.modular.com/mojo/changelog.html#section-1). Very excellent add!

> Mojo now supports packages! A Mojo package is defined by placing an __init__.mojo or __init__.🔥 within a directory. Other files in the same directory form modules within the package, this works exactly like it does in Python, for example...

1. This does not seem to be mentioned in the mojo docs, or programming manual? Seems like a useful thing to add!
2. The jupyter playground does not seem to facilitate really creating and editing mojo files. I did manage to create a .text file and rename it as .mojo, and got syntax hiliting after that.
3.  Importing does not seem to work. I noticed with this in the playground. Not sure I am missing something obvious. Or should I file a bug?

Here is a silly struct which works OK defined in a notebook:

<img width="1072" alt="Screenshot 2023-08-02 at 1 53 55 AM" src="https://github.com/modularml/mojo/assets/766758/0c86f970-caff-40ef-86ea-4dad51a31009">

However, when importing the same struct from a mojo package, it raises some errors:

<img width="1660" alt="Screenshot 2023-08-02 at 1 56 42 AM" src="https://github.com/modularml/mojo/assets/766758/452c3329-284e-4e89-8881-56f05bee816b">

**Comment by mojodojodev** [comment, marked as answer]
At the moment you need to explicitly import things in the `.mojo` file that are usually imported by default:
```python
from SIMD import Float32
```
This issue was raised and closed as completed just today: https://github.com/modularml/mojo/issues/469

That usually means that it'll be in the next release, should be in the next few days.

**Reply by guidorice** [reply]
thanks @mojodojodev that worked.
